library/repair_docker_data_volumes

```diff
@@ -0,0 +1,338 @@
+            continue
+
+        # This config should turn into:
+        #
+        #   "MountPoints": {
+        #       "/opt/some/volume": {
+        #           "Destination": "/opt/some/volume",
+        #           "Driver": "local",
+        #           "Name": "33ff....e78b",
+        #           "Named": false,
+        #           "Propagation": "",
+        #           "RW": true,
+        #           "Relabel": "",
+        #           "Source": ""
+        #       }
+        #       "/opt/mapped/volume": {
+        #           "Destination": "/opt/mapped/volume",
+        #           "Driver": "",
+        #           "Name": "",
+        #           "Named": false,
+        #           "Propagation": "rprivate",
+        #           "RW": false,
+        #           "Relabel": "",
+        #           "Source": "/opt/mapped/volume/outside"
+        #       }
+        #   },
+        #
+        mountpoints = {}
+        for internal_path, external_path in volumes.iteritems():
+            rw = volumes_rw[internal_path]
+
+            match_result = (VFS_MOUNT_REGEX.match(external_path)
+                            or VOLUME_MOUNT_REGEX.match(external_path))
+            if match_result:
+                # unmapped volume
+                volume_id = match_result.group(1)
+                mountpoints[internal_path] = {
+                    'Destination': internal_path,
+                    'Driver': 'local',
+                    'Name': volume_id,
+                    'Named': False,
+                    'Propagation': '',
+                    'RW': rw,
+                    'Relabel': '',
+                    'Source': ''
+                }
+            else:
+                # mapped volume
+                mountpoints[internal_path] = {
+                    'Destination': internal_path,
+                    'Driver': '',
+                    'Name': '',
+                    'Named': False,
+                    'Propagation': 'rprivate',
+                    'RW': rw,
+                    'Relabel': '',
+                    'Source': external_path
+                }
+        config['MountPoints'] = mountpoints
+
+        backup_path = os.path.join(container_path, BACKUP_FILENAME)
+        yield RewriteConfigOperation(config_v2_path, backup_path, json.dumps(config))
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            fix_vfs_links=dict(required=True, type='bool'),
+            fix_container_mountpoints=dict(required=True, type='bool'),
+            old_container_configs=dict(required=True)),
+        supports_check_mode=True)
+
+    # We need to figure out what to do before doing it.
+    # That's because, when we execute our operations, we shut down the docker daemon first -
+    # but figuring out what to do, in some cases, requires us to query the docker daemon.
+
+    operations = []
+    if module.params['fix_vfs_links']:
+        operations += list(fix_volume_mount_missing_symlinks(module))
```

Comment:
Or perhaps `operations.extend(fix_volume_mount_missing_symlinks(module))`?

That uses the generator directly instead of constructing a list only for the purpose of concatenating it with the previous contents.